v2/dtos/common/base.go

```diff
@@ -14,6 +18,10 @@ type BaseRequest struct {
 	RequestId string `json:"requestId" validate:"len=0|uuid"`
 }
 
+func NewBaseRequest() BaseRequest {
+	return BaseRequest{RequestId: uuid.New().String()}
```

Comment:
we need ApiVersion more than RequestId here

Reply:
BaseRequest currently doesn't contain ApiVersion, the sub-DTOs (Event & Reading) do contain the ApiVersion. I will add ApiVersion to BaseRequest so it is consistent with BaseResponse and sub-DTOs.